openapi.json

```diff
@@ -0,0 +1,9995 @@
+            "title": "matching_status",
+            "type": "string",
+            "x-nullable": true
+          },
+          "person_id": {
+            "description": "to be documented",
+            "title": "person_id",
+            "type": "integer"
+          },
+          "role": {
+            "description": "to be documented",
+            "enum": [
+              "Commercial Card Holder",
+              "Beneficial Owner",
+              "Company Owner",
+              "Trustee",
+              "Settlor",
+              "Grantor",
+              "Founder",
+              "Beneficiary",
+              "Power of Attorney",
+              "Authorised Signatory",
+              "Purchasing Account",
+              "others"
+            ],
+            "title": "role",
+            "type": "string"
+          },
+          "taggable_entity_id": {
+            "description": "to be documented",
+            "title": "taggable_entity_id",
+            "type": "integer"
+          }
+        },
+        "required": ["person_id", "role", "taggable_entity_id"],
+        "type": "object"
+      },
+      "tag": {
+        "example": {
+          "changed_by": "changed_by",
+          "changed_on": "2000-01-23T04:56:07+00:00",
+          "created_by": "created_by",
+          "created_on": "2000-01-23T04:56:07+00:00",
+          "description": "description",
+          "id": 0,
+          "tagged_entities": [6, 6]
+        },
+        "properties": {
+          "changed_by": {
+            "description": "User name/email of the user who last changed the record via API",
+            "readOnly": true,
+            "title": "changed_by",
+            "type": "string"
+          },
+          "changed_on": {
+            "description": "Timestamp on which the record was last changed, automatically filled by the API",
+            "format": "date-time",
+            "readOnly": true,
+            "title": "changed_on",
+            "type": "string"
+          },
+          "created_by": {
+            "description": "User name/email of the user who inserted the record via API",
+            "readOnly": true,
+            "title": "created_by",
+            "type": "string"
+          },
+          "created_on": {
+            "description": "Timestamp on which the record was created, automatically filled by the API",
+            "format": "date-time",
+            "readOnly": true,
+            "title": "created_on",
+            "type": "string"
+          },
+          "description": {
+            "description": "to be documented",
+            "nullable": true,
+            "title": "description",
+            "type": "string",
+            "x-nullable": true
+          },
+          "id": {
+            "description": "to be documented",
+            "readOnly": true,
+            "title": "id",
+            "type": "integer"
+          },
+          "tagged_entities": { "items": { "type": "integer" }, "type": "array" }
```

Comment:
Regarding tagged_entities, would you be open to replacing

```json
"tagged_entities": { "items": { "type": "integer" }, "type": "array" }
```

with
```

type tag {
  # ...
  tagged_entities: [tagged_entity!]
}

type person_role {
  # ...
  tagged_entity: tagged_entity!
}

type memo {
  # ...
  tagged_entity: tagged_entity!
}

type tagged_entity {
  entity_type: taggable_entity!
  id: Int!
}

enum taggable_entity {
  basic_account
  card
  master_account
  memo
  person
}
```

This change if possible, would add great clarification for us and quick access of exactly what endpoint to ask for the specific entity.

Reply:
(we don't know which endpoint to address if only a oid but no entity is known)

Reply:
@jukietz that’s for you to answer => could we return the exact entity the taggable_entity_id refers to?
I.e. is it a Master, Basic, Card or Person record?

Reply:
We could add an (get_by_id & search) API to taggle-entity, currently it has none

Reply:
This should work well for now 👍